### ISSUE TITLE
OneDrive For Business: Users from other tenants

### ISSUE THREAD

**Issue by DJAndries** [OneDrive/onedrive-sdk-android#104]
When authenticating OneDrive for Business users inside the same organization with the Azure AD application registration, everything works OK.

However, when users from other tenants attempt to authenticate, it fails to get the files service during discovery ("Unable to file the files services from the directory provider").

I noticed that when registering the app on Azure AD, there is no "Multi-Tenant" checkbox if a Native application is being registered. This option is available if registering a Web App. Could this be part of the problem?

**Comment by peternied** [comment]
@DJAndries If you follow the instructions here https://github.com/OneDrive/onedrive-api-docs/blob/master/app-registration.md#register-your-app-for-onedrive-for-business are they able to get you unblocked?

**Comment by DJAndries** [comment]
I did follow those instructions with no luck. Perhaps I should also note that I am using an Office 365 trial account to test the multi-tenant login

**Comment by peternied** [comment]
I found a question in the msdn community forums that has some additional detail in the same vein as your problem, give https://social.msdn.microsoft.com/Forums/en-US/2f821618-4818-458f-839b-d16c2372dde7/how-to-add-multitenant-support-to-an-office-365-sample-app?forum=appsforoffice a shot.

**Comment by DJAndries** [comment]
When registering a "Native" application on Azure AD, is it automatically Multi-Tenant? There is no option or switch for toggling the multi-tenant setting for Native apps.

When logging in, inside the ADALAuthenticator.getOneDriveApiService method, the array of services will contain other services (such as Calendar, Mail, Directory, etc.) but no 'MyFiles' or 'RootSite' elements. Those two elements normally appear when logging in with our organization AD accounts.

**Comment by DJAndries** [comment]
I have resolved the issue after receiving a support ticket from a customer who received this error message when trying to log in: "Calling principal cannot consent due to lack of permissions". After some searching, I found this article: https://social.msdn.microsoft.com/Forums/office/en-US/40da258e-d036-4376-bb1c-7f037a99d635/aadsts50020-calling-principal-cannot-consent?forum=exchangesvrdevelopment 

Devs in the forum thread fixed the issue by removing any Admin level permissions. I was adding permissions to the Azure AD registration using the old portal (which does not show the Admin level of each permission). I used to new portal to identify Admin only permissions that I have added and removed them. The Office 365 trial account now works OK
